direct/io-bulk-fs/src/main/java/cz/o2/proxima/direct/bulk/FileSystem.java

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright 2017-2020 O2 Czech Republic, a.s.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package cz.o2.proxima.direct.bulk;
+
+import cz.o2.proxima.annotations.Internal;
+import cz.o2.proxima.util.ExceptionUtils;
+import java.io.File;
+import java.io.Serializable;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+/** A proxima's abstraction of bulk FS. */
+@Internal
+public interface FileSystem extends Serializable {
+
+  static FileSystem local(File parent, NamingConvention convention) {
```

Comment:
Can we use `java.nio` here instead of the old API?

https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html

Reply:
I guess we can. The "problem" would be, that JUnit 4 and its TemporaryFolder returns the old API. I would leave this refactor as part of moving to JUnit 5, which can create temporary files using the new API. Otherwise we would just convert APIs back and forth, without any significant benefit.